catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -355,9 +358,42 @@ private String getExceptionMessage(String responseType, String previousException
         return builder.toString();
     }
 
+    public void shutdown() {
+        getPeriodicDrain().shutdown();
+    }
+
+    public String getRootBackupDir() {
+        return rootBackupDir.getAbsolutePath();
+    }
+
+    /**
+     * Sets the root file system backup directory.
+     *
+     * @param dir absolute path for the root file system backup directory.
+     */
+    public void setRootBackupDir(String dir) {
+        if (StringUtils.isBlank(dir)) {
+            LOGGER.error("The root backup directory is blank.");
```

Comment:
We should throw an exception. We should also validate things a little bit more, i.e., is the path valid, can the directory be created and written to, etc. That way we can assume the path is valid everywhere else (no need for `null` checks) and admins get immediate feedback when they set the value in the UI.